src/polyline2_signed_distance_constraint.cpp

```diff
@@ -102,30 +106,32 @@ void Polyline2SignedDistanceConstraint::Quadraticize(const VectorXf& input,
   const float dy2 = dy * dy;
   const float level = orientation * (signed_threshold_sq_ - signed_distance_sq);
 
-  // Handle cases separately depending on whether or not closest point is
-  // a vertex of the polyline.
-  if (!is_vertex) {
-    const Point2 relative = current_position - segment.FirstPoint();
-    const Point2& unit_direction = segment.UnitDirection();
-    const float cross =
-        relative.x() * unit_direction.y() - relative.y() * unit_direction.x();
-    CHECK_EQ(sgn(cross), sgn(signed_distance_sq));
-
-    const float coeff = 2.0 * orientation * sign / level;
-    const float grad_coeff = coeff * cross;
-    const float weighted_grad_coeff = weight_ * grad_coeff;
-    (*grad)(xidx_) += weighted_grad_coeff * unit_direction.y();
-    (*grad)(yidx_) -= weighted_grad_coeff * unit_direction.x();
-
-    const float hess_coeff = weight_ * coeff * (grad_coeff * cross + 1.0);
-    const float hess_xy = hess_coeff * unit_direction.x() * unit_direction.y();
-    (*hess)(xidx_, xidx_) +=
-        hess_coeff * unit_direction.y() * unit_direction.y();
-    (*hess)(yidx_, yidx_) +=
-        hess_coeff * unit_direction.x() * unit_direction.x();
-    (*hess)(xidx_, yidx_) -= hess_xy;
-    (*hess)(yidx_, xidx_) -= hess_xy;
-  } else {
+  // First check if closest point is an endpoint of the polyline
+  if(!is_endpoint){
```

Comment:
nit: spaces before and after parens, i.e. `if (!is_endpoint) {`

Reply:
done!